### ISSUE TITLE
[java][gradle] script environment can be ignored for built-artifacts with default actions

### ISSUE THREAD

**Issue by javabrett** [travis-ci/travis-ci#8937]
# TL;DR

The environment specified for `script` via `before_script` is not active for the actual build of the project artifacts (`*.jar`) when running with default Java+Gradle actions.

# Details

I have a trivial Java/Gradle project which seems to highlight shortcomings in the _default_ actions when Travis detects Gradle.  The current default actions/scripts are:

```
install:
  - ./gradlew assemble`
script:
  - ./gradlew check
```

There's nothing exotic about the Java/Gradle project - it does perform some simple `processResources` resource-filtering based on the build environment, and makes a test-assertion on that resource-filtering.

`.travis.yml`:

```
dist: trusty
sudo: false
language: java

env:
  - ENVIRONMENT=
  - ENVIRONMENT=dev
  - ENVIRONMENT=prod

before_script:
  - export ENV=$ENVIRONMENT && echo ENV:$ENV

after_script:
  - unzip -c build/libs/*.jar main.properties
```
The only perhaps-contrived piece being the global `ENVIRONMENT` copied to `ENV` in `before_script` for use in `script` only.

The results are:

Outside Travis:

* `/usr/bin/env ENV= ./gradlew clean check # PASS`
* `/usr/bin/env ENV=dev ./gradlew clean check # PASS`
* `/usr/bin/env ENV=prod ./gradlew clean check # PASS`

Travis:
```
ENVIRONMENT= # PASS
ENVIRONMENT=dev # FAIL
Gradle suite > Gradle test > MainTest.testGetEnv FAILED
    java.lang.AssertionError: unexpected ENV expected [dev] but found []
ENVIRONMENT=prod # FAIL
Gradle suite > Gradle test > MainTest.testGetEnv FAILED
    java.lang.AssertionError: unexpected ENV expected [prod] but found []
```

I know what causes this (and accept that it can be remedied by customising the build steps), but I wanted to get some feedback on whether this can be considered a bug (in the default actions) or not.  The docs state:

> A build on Travis CI is made up of two steps:
> 1. install: install any dependencies required
> 2. script: run the build script

... however it can be shown that the build script does not in-fact build the project's output artifacts here or by-default.  Therefore the `script` environment is being ignored.
